src/stores/WalletStore.ts

```diff
@@ -158,13 +155,27 @@ class WalletStore extends PersistableStore {
       void this.handleAccountChanged()
     })
     provider.on('disconnect', (error: unknown) => {
+      // Sometimes this error fires when user switching between networks too fast, we should not clean data in this case
+      if (
+        error instanceof Error &&
```

Comment:
I think we can extract his into `helpers`, maybe we have something like this in `handleErro`

Reply:
Nice idea, ty